### PR TITLE
feat!: Activate decimal by default

### PR DESCRIPTION
**Pull request by ritchie46**

### CODE REVIEW

**Comment by stinodego** [comment]
@ritchie46 this is already done: 
https://github.com/pola-rs/polars/pull/15324

**Comment by ritchie46** [comment]
Ah, you beat me to it. I do think we need the `activate-decimal` panic.

**Comment by stinodego** [comment]
> Ah, you beat me to it. I do think we need the `activate-decimal` panic.

Right, that is indeed nice for usability.

I chose deprecate `activate_decimals` and turn it into a no-op. But if you prefer to remove it entirely that's also fine.

File: crates/polars-core/src/series/from.rs [hunk 1]
```diff
@@ -444,30 +443,38 @@ impl Series {
...
-                    .into_decimal_unchecked(precision, scale)
-                    .into_series())
-            },
-            #[allow(unreachable_patterns)]
-            ArrowDataType::Decimal256(_, _) | ArrowDataType::Decimal(_, _) => {
-                if verbose() {
-                    eprintln!(
-                        "Polars does not support decimal types so the 'Series' are read as Float64"
-                    );
+                #[cfg(not(feature = "dtype-decimal"))]
```
Comment (ritchie46) [inline crates/polars-core/src/series/from.rs:448]: THIS

**Comment by ritchie46** [comment]
> I chose deprecate activate_decimals and turn it into a no-op. But if you prefer to remove it entirely that's also fine.


No, I mean this code, look for "THIS"` ^^


Otherwise, the `decimal` arrow datatype will be skipped and you get an error that decimals are not supported if you don't activate the decimal type.

**Comment by stinodego** [comment]
Yes I know, the two comments were unrelated 😄

**Comment by ritchie46** [comment]
Yeah, I like cleaning it up. ^^ It is 1.0 :D

**Review by stinodego — APPROVED** [review]

**Pull request merged**